src/uhs/transaction/validation.cpp

```diff
@@ -419,4 +456,115 @@ namespace cbdc::transaction::validation {
                                    && tx.verify(secp_context.get(), att);
                            });
     }
+
+    auto check_mint_witness(
```

Comment:
These mint witness check functions seem almost identical to the existing functions for input witnesses. Can we refactor them somehow?

Here and below.